### ISSUE TITLE
"Duplicate tag keys found." on stack update

### ISSUE THREAD

**Issue by MickVanDuijn** [serverless/serverless#7885]
This issue seems to be related to a recent [update to CloudFormation itself](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/ReleaseHistory.html) on June 22, 2020.
`Use the Tags property to add tag keys and values to ....`

When creating/updating a stack with a new resource (an IAM role in my case), and having a custom `stage` tag, CloudFormation will try to use the `stackTags` to create some of the resources it's creating under the hood (IAM roles for example). `serverless` framework is injecting a default `STAGE` tag, but due to NodeJS being case sensitive, the `stage` and `STAGE` tags are not merged. This will result in an error during CloudFormation updates: `Duplicate tag keys found. Please note that Tag keys are case insensitive. (Service: AmazonIdentityManagement; Status Code: 400; Error Code: InvalidInput; Request ID: xxxxxx-xxxxx)`

<details>
<summary><code>serverless.yml</code></summary>

```yaml
service: example-service

# NOTE: first deploy without this resource, then add it back in
resources:
  Resources:
    ExampleRole:
      Type: AWS::IAM::Role
      Properties:
        RoleName: ExampleRole
        AssumeRolePolicyDocument:
          Version: '2012-10-17'
          Statement:
            - Effect: Allow
              Principal:
                Service:
                  - lambda.amazonaws.com
              Action: sts:AssumeRole
        ManagedPolicyArns:
          - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole

provider:
  name: aws
  runtime: nodejs12.x
  region: eu-west-1
  stage: ${opt:stage, 'dev'}
  stackTags:
    my_tag: foo
    my_other_tag: bar
    stage: ${self:provider.stage}

```

</details>

<details>
<summary><b><code>serverless deploy</code> output</b></summary>

```
Serverless: Packaging service...
Serverless: Uploading CloudFormation file to S3...
Serverless: Uploading artifacts...
Serverless: Validating template...
Serverless: Updating Stack...
Serverless: Checking Stack update progress...
....
Serverless: Operation failed!
Serverless: View the full error output: https://eu-west-1.console.aws.amazon.com/cloudformation/home?region=eu-west-1#/stack/detail?stackId=arn%3Aaws%3Acloudformation%3Aeu-west-1%xxxxxxxxxxxxxxxx
 
  Serverless Error ---------------------------------------
 
  An error occurred: ExampleRole - Duplicate tag keys found. Please note that Tag keys are case insensitive. (Service: AmazonIdentityManagement; Status Code: 400; Error Code: InvalidInput; Request ID: xxxxxx-xxxxxxxx).

```

</details>

<b>Installed version</b>

```
Framework Core: 1.74.0
Plugin: 3.6.13
SDK: 2.3.1
Components: 2.31.4
```
## Solution

This line https://github.com/serverless/serverless/blob/f015abfd11b11a12e24e70fd32a32de0f2ffb193/lib/plugins/aws/lib/updateStack.js#L21 doesn't perform a case-insensitive merge of the `stackTags`, that are initialized with `STAGE` by default https://github.com/serverless/serverless/blob/f015abfd11b11a12e24e70fd32a32de0f2ffb193/lib/plugins/aws/lib/updateStack.js#L14 

Due to the standards we use, we prefer to keep the casing of `stage` the way we have it.

The solution would be to perform a case-insensitive merge of the objects, **however**, it would be important that the "newly added keys" take precedence over the existing default key(s).

**Comment by medikoo** [comment]
@MickVanDuijn great thanks for describing the issue. PR with a fix is definitely welcome!

**Comment by Vadorequest** [comment]
Encountered this issue using `1.74.0`, fixed by upgrading to `1.79.0`.
